docs/control-structures.rst

```diff
@@ -322,14 +322,16 @@ In the following example, we show how ``throw`` can be used to easily revert an
         }
     }
 
-Currently, there are six situations, where exceptions happen automatically in Solidity:
+Currently, there are situations, where exceptions happen automatically in Solidity:
 
-1. If you access an array beyond its length (i.e. ``x[i]`` where ``i >= x.length``).
-2. If a function called via a message call does not finish properly (i.e. it runs out of gas or throws an exception itself).
-3. If a non-existent function on a library is called or Ether is sent to a library.
+1. If you access an array at a too large or negative index (i.e. ``x[i]`` where ``i >= x.length`` or ``i < 0``).
+2. If you access a fixed-length ``bytesN`` at a too large or negative index.
+3. If you call a function via a message call but it does not finish properly (i.e. it runs out of gas, has no matching function, or throws an exception itself), except when a low level operation ``call``, ``send``, ``delegatecall`` or ``callcode`` is used.  The low level operations never throw exceptions but indicate failures by return values being ``false``.
 4. If you divide or modulo by zero (e.g. ``5 / 0`` or ``23 % 0``).
 5. If you perform an external function call targeting a contract that contains no code.
-6. If a contract-creation call using the ``new`` keyword fails.
+6. If you call a contract-creation using the ``new`` keyword but it does not finish properly.
```

Comment:
`If a contract creation (using the ``new`` keyword) does not finish properly or throws an exception itself`